### ISSUE TITLE
VM scaleset : BLOCKED support enabling auto-scale

### ISSUE THREAD

**Issue by BurtBiel** [Azure/azure-cli#815]
This is currently not available in any SDK, but it may be at some point.

**Comment by yugangw-msft** [comment]
It is not blocked, rather we need to use resource `Microsoft.Insights/autoscaleSettings` to define the rule. Suggest to expose a new command `auto-scale` for this. I prefer not integrate with `create`, because it has a whole lot of arguments already

**Comment by tjprescott** [comment]
@vishrutshah I will handle the integration into the create command--just need to know what the good "default" scaleset settings should be. I think we should use the same default as your `vmss autoscale set` command.

**Comment by vishrutshah** [comment]
@tjprescott Awesome! Based on what I see in the Azure portal and Azure gallery template, the default should look like below for `vmss autoscale`:

```
{
  "autoscale_setting_resource_name": "{MyAutoscaleSettings}",
  "location": "{location}",
  "tags": {},
  "profiles": [
    {
      "name": "Profile1",
      "capacity": {
        "minimum": "1",
        "maximum": "10",
        "default": "1"
      },
      "rules": [
        {
          "metric_trigger": {
            "metric_name": "Percentage CPU",
            "metric_resource_uri": "/subscriptions/{subscriptionID}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/virtualMachineScaleSets/{VMSSName}",
            "time_grain": "PT1M",
            "statistic": "Average",
            "time_window": "PT5M",
            "time_aggregation": "Average",
            "operator": "GreaterThan",
            "threshold": 75
          },
          "scale_action": {
            "direction": "Increase",
            "type": "ChangeCount",
            "value": "1",
            "cooldown": "PT1M"
          }
        },
        {
          "metric_trigger": {
            "metric_name": "Percentage CPU",
            "metric_resource_uri": "/subscriptions/{subscriptionID}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/virtualMachineScaleSets/{VMSSName}",
            "time_grain": "PT1M",
            "statistic": "Average",
            "time_window": "PT5M",
            "time_aggregation": "Average",
            "operator": "LessThan",
            "threshold": 25
          },
          "scale_action": {
            "direction": "Decrease",
            "type": "ChangeCount",
            "value": "1",
            "cooldown": "PT1M"
          }
        }
      ]
    }
  ],
  "notifications": [],
  "enabled": true,
  "target_resource_uri": "/subscriptions/{subscriptionID}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/virtualMachineScaleSets/{VMSSName}"
}
```

Basically one profile with two rules. One for scale out and one for scale in. 

Let me know if you need any more information on this!

Thanks!

**Comment by yugangw-msft** [comment]
//cc: @gbowerman @gatneil 
We will have min support from the new monitoring commands to be in par with xplat, and this issue is for convenient command more towards VMSS

**Comment by gbowerman** [comment]
Looks reasonable #sign-off

**Comment by gatneil** [comment]
Agreed :)

**Comment by gatneil** [comment]
Actually, one thought:

@yugangw-msft you say "We will have min support from the new monitoring commands to be in par with xplat"; I didn't think CLI1.0 had much w/ respect to autoscale; looking at this doc: https://docs.microsoft.com/en-us/azure/monitoring-and-diagnostics/insights-cli-samples

it seems that you need to pass in the autoscale settings spec as a json string or create the settings object first via some other method then reference it from CLI. Is this the goal for parity? Ideally a lot of the json block for autoscale could have default values provided by CLI2.0 with the ability to override values as necessary (just like the vmss block does). Granted, if this issue is specifically for helpful autoscale commands within the vmss context, then this point is perhaps out of scope, but I figure a lot of the work for making the vmss autoscale commands easy could be re-used for autoscale commands on other autoscalable resources as well (websites, etc.).

**Comment by yugangw-msft** [comment]
> I figure a lot of the work for making the vmss autoscale commands easy could be re-used for autoscale commands on other autoscalable resources as well (websites, etc.).

@gatneil  Agreed.  I thought about this as well this morning. When I get to the scale support for website, I might refactor a bit for common supports. At current moment, let us focus on getting the autoscale right for vmss. The vmss is all about scale, and it deserves to have its own strong type commands more than any other ones.

**Comment by tjprescott** [comment]
The plan we discussed at the monitor commands review was to expose the "generic" autoscale commands under monitor as a stop-gap, but to expose commands in each applicable area (web app, vmss, etc) that would have a tailored experience.

**Comment by tjprescott** [comment]
Pushing this to next sprint while I await recommendations on the most common autoscale scenarios for VMSS.

**Comment by gbowerman** [comment]
@tjprescott who is this waiting on?

**Comment by tjprescott** [comment]
Monitor team will be providing templates for the common VMSS autoscale patterns. The convenience command will be tailored to those scenarios (which are more than just the CPU autoscale that the portal exposes today).

**Comment by yugangw-msft** [comment]
//CC: @mayurid  @troydai 
Once monitoring support gets consolidated, please handle this and related [webapp ask](https://github.com/Azure/azure-cli/issues/1834)
